src/components/legend.ts

```diff
@@ -280,6 +282,7 @@ export module Components {
       entries.select("path").attr("d", (d: any, i: number, j: number) => this.symbol()(d, j)(layout.textHeight * 0.6))
                             .attr("transform", "translate(" + (layout.textHeight / 2) + "," + layout.textHeight / 2 + ")")
                             .attr("fill", (value: string) => this._colorScale.scale(value))
+                            .attr("opacity", (d: any, i: number, j: number) => this.symbolOpacity()(d, j))
```

Comment:
Yes. I'm still not sure how the indices are being passed around, but when I wrote up the index unit tests for symbolOpacity() in the same style as symbol(), I realized the index was off and doing the same thing here with the j solved it.